tasks/create_app_vault.yml

```diff
@@ -5,27 +5,32 @@
   debug: msg="App name {{ app.name }}"
   tags: vault
 
-- name: retrieve template file
-  local_action:
-    module: stat
-    path: "apps/{{ app.name }}/vars/vault/main.yml.j2"
+- name: Register the template path
+  command: echo  "apps/{{ app.name }}/vars/vault/main.yml.j2"
+  register: vault_template_path
+  no_log: true
+
+- name: set vault template path variable
+  set_fact:
+    vault_template_path: "{{ vault_template_path.stdout }}"
+
+- name: Retrieve template file
+  stat:
+    path: "{{ vault_template_path }}"
   register: vault_template
+  no_log: true
 
-- name: check if template file exists
+- name: Check if template file exists
   fail:
-    msg: "file apps/{{ app.name }}/vars/vault/main.yml.j2 does not exists"
+    msg: "file {{ vault_template_path }} does not exists"
   when: vault_template.stat.exists == False
 
-
-- name: execute template to create vault file
-  local_action:
-    module: template
-    src: "{{ item }}"
+- name: Create application vault
+  template:
+    src: "{{ vault_template_path }}"
     dest: "group_vars/customer/{{ customer }}/{{ env_type }}/secrets/{{ app.name }}.vault.yml"
-  with_items:
-    - "apps/{{ app.name }}/vars/vault/main.yml.j2"
 
-- name: encrypt newly create vault file
-  local_action: command ansible-vault encrypt {{ item }}
+- name: Encrypt newly create vault file
+  command: ansible-vault encrypt {{ item }}
   with_items:
     - "group_vars/customer/{{ customer }}/{{ env_type }}/secrets/{{ app.name }}.vault.yml"
```

Comment:
As you are using it multiple times, define a variable for the application's vault path.